tokio/src/runtime/handle.rs

```diff
@@ -201,6 +201,36 @@ impl Handle {
         let _ = self.blocking_spawner.spawn(task, &self);
         handle
     }
+
+    /// TODO: write docs if this is a good direction
+    pub fn block_on<F: Future>(&self, future: F) -> F::Output {
+        // Enter the **runtime** context. This configures spawning, the current I/O driver, ...
+        let _rt_enter = self.enter();
+
+        // Enter a **blocking** context. This prevents blocking from a runtime.
+        let mut _blocking_enter = crate::runtime::enter(true);
+
+        // Block on the future
+        _blocking_enter
+            .block_on(future)
+            .expect("failed to park thread")
+    }
+
+    cfg_io_driver! {
```

Comment:
Should this `cfg_` just be inside the `fn shutdown` around the `if let`?

Reply:
I did try that at first but the macro takes [an `item`](https://github.com/EmbarkStudios/tokio/blob/rt-handle-block-on/tokio/src/macros/cfg.rs#L65) so can't easily be put into functions afaik. It also checks [quite a few features](https://github.com/EmbarkStudios/tokio/blob/rt-handle-block-on/tokio/src/macros/cfg.rs#L64-L80) so didn't feel right copy pasting that.